src/Elastic.Markdown/Slices/HtmlWriter.cs

```diff
@@ -12,29 +14,67 @@ public class HtmlWriter(DocumentationSet documentationSet, IFileSystem writeFile
 {
 	private DocumentationSet DocumentationSet { get; } = documentationSet;
 
-	private async Task<string> RenderNavigation(MarkdownFile markdown, Cancel ctx = default)
+	private async Task<string> RenderNavigation(ConfigurationFile configuration, string topLevelGroupId, MarkdownFile markdown, Cancel ctx = default)
 	{
+		var group = DocumentationSet.Tree.NavigationItems
+			.OfType<GroupNavigation>()
+			.FirstOrDefault(i => i.Group.Id == topLevelGroupId)?.Group;
+
 		var slice = Layout._TocTree.Create(new NavigationViewModel
 		{
-			Tree = DocumentationSet.Tree,
-			CurrentDocument = markdown
+			Title = group?.Index?.NavigationTitle ?? DocumentationSet.Tree.Index!.NavigationTitle!,
+			TitleUrl = group?.Index?.Url ?? DocumentationSet.Tree.Index!.Url,
+			Tree = group ?? DocumentationSet.Tree,
+			CurrentDocument = markdown,
+			IsRoot = topLevelGroupId == DocumentationSet.Tree.Id,
+			Configuration = configuration
 		});
 		return await slice.RenderAsync(cancellationToken: ctx);
 	}
 
-	private string? _renderedNavigation;
+	private readonly Dictionary<string, string> _renderedNavigationCache = [];
 
 	public async Task<string> RenderLayout(MarkdownFile markdown, Cancel ctx = default)
 	{
 		var document = await markdown.ParseFullAsync(ctx);
 		return await RenderLayout(markdown, document, ctx);
 	}
 
+	private static string GetTopLevelGroupId(MarkdownFile markdown) =>
+		markdown.YieldParentGroups().Length > 1
+			? markdown.YieldParentGroups()[^2]
+			: markdown.YieldParentGroups()[0];
+
 	public async Task<string> RenderLayout(MarkdownFile markdown, MarkdownDocument document, Cancel ctx = default)
 	{
-		var html = MarkdownFile.CreateHtml(document);
+		var html = markdown.CreateHtml(document);
 		await DocumentationSet.Tree.Resolve(ctx);
-		_renderedNavigation ??= await RenderNavigation(markdown, ctx);
+
+		var topLevelNavigationItems = DocumentationSet.Tree.NavigationItems
+			.OfType<GroupNavigation>()
+			.Select(i => i.Group);
+
+		string? navigationHtml;
+
+		if (DocumentationSet.Configuration.Features.ContainsKey("primary-nav"))
```

Comment:
Lets have a static class with constant string properties so we can use `Features.PrimaryNavigation`.

Reply:
https://github.com/elastic/docs-builder/pull/604/commits/ee10395c27ec65fd6c11127bacec42c344aa0a80

Instead of a static constant I created a class. WDYT?